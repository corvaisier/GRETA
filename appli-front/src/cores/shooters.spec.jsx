import React from 'react';
import { render, fireEvent, screen, act } from '@testing-library/react'

import Shooter from './shooters'

describe('<Shooters />', () => {
    it('Should display error message for negative number', async () => {
        render(<Shooter />);
        fireEvent.change(document.querySelector('.form-control'), { target: {value: "-3" }} );
        await act(async () => {
            fireEvent.click(document.querySelectorAll('input')[1])
        });
        screen.getByText("Entrez s'il vous plaît une valeur intégrale comprise entre 0 ett 99")
        expect(document.body).toMatchSnapshot();
    })

})