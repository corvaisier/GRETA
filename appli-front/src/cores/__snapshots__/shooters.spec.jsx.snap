// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shooters /> Should display error message for negative number 1`] = `
<body>
  <div>
    <div
      class="container"
      style="text-align: center; margin: 30px;"
    >
      <form
        class="form-group"
      >
        <label
          for="formGroupExampleInput"
        >
          Nombre de shooters:
          <input
            class="form-control"
            id="formGroupExampleInput"
            type="text"
            value="-3"
          />
        </label>
        <input
          type="submit"
          value="Envoyer"
        />
        <div>
          Entrez s'il vous plaît une valeur intégrale comprise entre 0 et 99
        </div>
      </form>
    </div>
  </div>
</body>
`;
